src/main/java/com/epam/msu/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+package com.epam.msu.service.impl;
+
+import com.epam.msu.dao.impl.CertificateDaoImpl;
+import com.epam.msu.dao.impl.TagDaoImpl;
+import com.epam.msu.dto.CertificateDto;
+import com.epam.msu.entity.Certificate;
+import com.epam.msu.entity.Tag;
+import com.epam.msu.service.CertificateService;
+import com.epam.msu.util.MappingUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CertificateServiceImpl implements CertificateService {
+
+    private final CertificateDaoImpl certificateDao;
+    private final TagDaoImpl tagDao;
+
+    @Autowired
+    public CertificateServiceImpl(CertificateDaoImpl certificateDao, TagDaoImpl tagDao) {
+        this.certificateDao = certificateDao;
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    public List<CertificateDto> getAllCertificates() {
+        List<Certificate> certificateList = certificateDao.getAllCertificates();
+        List<CertificateDto> certificatesDto = MappingUtils.mapToListCertificateDto(certificateList);
+        List<Tag> tags = new ArrayList<>();
+        for (CertificateDto certificate : certificatesDto) {
+            List<Tag> tag = tagDao.getTagsByCertificateId((int) certificate.getId());
+            certificate.setTag(tag);
+
+        }
+        ;
+        return certificatesDto;
+    }
+
+    @Override
+    public CertificateDto getCertificateById(int id) {
+        Certificate certificate = certificateDao.getCertificateById(id);
```

Comment:
It is better to throw exception in case of certificate not found instead of returning null

Reply:
fixed